### PR TITLE
Restructure documentation and improve TOC.

### LINKED ISSUE #8661: [DOC] Some ideas for the readthedocs style
We should iterate on our style for readthedocs. The style of the brownie documentation looks nice: https://eth-brownie.readthedocs.io/en/latest/

Two items I especially like about that:
 - second level in the TOC is expanded by default
 - it has a "dark mode"

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/8661

### CODE REVIEW

**Comment by chriseth** [comment]
image

**Comment by chriseth** [comment]
I'm sorry that this is hard to review. What I did: Split up misc into individual files. "Inline" the "Solidity in Depth" file directly into the main TOC and rearrange some items in the main TOC.

File: docs/index.rst [hunk 1]
```diff
@@ -36,7 +36,7 @@ If you are new to the concept of smart contracts we recommend you start with
 :ref:`an example smart contract <simple-smart-contract>` written
 in Solidity. When you are ready for more detail, we recommend you read the
 :doc:`"Solidity by Example" <solidity-by-example>` and
-:doc:`"Solidity in Depth" <solidity-in-depth>` sections to learn the core concepts of the language.
+"Language Description" sections to learn the core concepts of the language.
```
Comment (chriseth) [inline docs/index.rst:39]: Had to change this like because "Solidity in Depth" cannot be linked to anymore.

File: docs/miscellaneous.rst [hunk 2]
```diff
@@ -1,814 +0,0 @@
...
- - If a ``:`` is missing, all following fields are considered empty.
-
-This means the following source mappings represent the same information:
-
-``1:2:1;1:9:1;2:1:2;2:1:2;2:1:2``
-
-``1:2:1;:9;2:1:2;;``
-
-***************
-Tips and Tricks
```
Comment (leonardoalt) [inline docs/miscellaneous.rst:602]: Is this anywhere now?

**Review by leonardoalt — CHANGES_REQUESTED** [review]

File: docs/miscellaneous.rst (hunk 2 above)
Comment (chriseth) [reply docs/miscellaneous.rst:602]: No, I did not know where to properly put this and it is not too useful anyway...

File: docs/internals/variable_cleanup.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
+.. index: variable cleanup
+
+*********************************
+Internals - Cleaning Up Variables
```
Comment (leonardoalt) [inline docs/internals/variable_cleanup.rst]:
```suggestion
Cleaning Up Variables
```
To be consistent with the others (check the image you pasted)

**Review by leonardoalt — CHANGES_REQUESTED** [review]

File: docs/internals/variable_cleanup.rst (hunk 3 above)
Comment (chriseth) [reply docs/internals/variable_cleanup.rst]: fixed.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**